pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go

```diff
@@ -211,6 +217,23 @@ func installer(allWorkers bool, secretName string) []runtime.Object {
 			},
 		},
 	}
+
+	if generation != "0" {
+		plan.Spec.Secrets = append(plan.Spec.Secrets, upgradev1.SecretSpec{
+			Name: generationSecretName,
+		})
+		objs = append(objs, &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generationSecretName,
+				Namespace: namespaces.System,
+			},
+			StringData: map[string]string{
+				"generation": generation,
+			},
+		})
+	}
+
+	return append(objs, plan)
```

Comment:
Previous code has `plan` as the first object in the result, this one would have it as last. This may not change anything functionally, but ideally we want to preserve the same order.

Reply:
:+1: